libraries/AP_Filesystem/AP_Filesystem.h

```diff
@@ -110,6 +110,10 @@ class AP_Filesystem {
     struct dirent *readdir(DirHandle *dirp);
     int closedir(DirHandle *dirp);
 
+    // return number of bytes that should be written before fsync for optimal
+    // streaming performance/robustness. if zero, any number can be written.
+    uint32_t bytes_until_fsync(int fd);
```

Comment:
Is there some reason all of these can't be `const`?

Reply:
This would not let implementations take member semaphores, which LittleFS currently does. So it's a nice idea but not feasible.